src/components/proofs/ReadyERC721ProofsList.tsx

```diff
@@ -1,25 +1,22 @@
 import { useSnapshot } from 'valtio'
 import ContractListContainer from 'components/proofs/ContractListContainer'
+import ERC721Proof from 'helpers/proofs/ERC721Proof'
 import Network from 'models/Network'
 import Proof from 'components/proofs/Proof'
 import ProofStore from 'stores/ProofStore'
-import networkPick from 'helpers/networkPick'
 
 export default function ({ network }: { network: Network }) {
-  const { goerliERC721ProofsCompleted, mainnetERC721ProofsCompleted } =
-    useSnapshot(ProofStore)
+  const { proofsCompleted } = useSnapshot(ProofStore)
 
-  const ERC721ProofsCompleted = networkPick(
-    network,
-    goerliERC721ProofsCompleted,
-    mainnetERC721ProofsCompleted
-  )
+  const allERC721ProofsByNetwork = proofsCompleted.filter(
+    (proof) => proof instanceof ERC721Proof && proof.network === network
+  ) as ERC721Proof[]
 
   return (
     <>
-      {!!ERC721ProofsCompleted?.length && (
+      {!!allERC721ProofsByNetwork?.length && (
```

Comment:
Why we need to do `!!` here? If the `length` is `0` or `allERC721ProofsByNetwork is undefined`, then it won't return anything, isn't it?